src/attributes.jl

```diff
@@ -1856,6 +1856,36 @@ function attribute_value_type(::ConstraintConflictStatus)
     return ConflictParticipationStatusCode
 end
 
+"""
+    UserDefinedFunction(name::Symbol, N::Int)
+
+Set a user-defined function by the name of `Name` with `N` arguments.
+
+The value to be set is a tuple containg one to three functions, representing the
+zero, first, and second-order derivatives of the function.
+
+## Examples
+
+```julia
+f(x, y) = x^2 + y^2
+function ∇f(g, x, y)
+    g .= 2 * x, 2 * y
+    return
+end
+function ∇²f(H, x...)
+    H[1, 1] = H[2, 2] = 2.0
+    return
+end
+MOI.set(model, MOI.UserDefinedFunction(:f, 2), (f,))
+MOI.set(model, MOI.UserDefinedFunction(:f, 2), (f, ∇f))
+MOI.set(model, MOI.UserDefinedFunction(:f, 2), (f, ∇f, ∇²f))
+```
+"""
+struct UserDefinedFunction <: AbstractModelAttribute
+    name::Symbol
+    N::Int
```

Comment:
The name is a bit odd, when all the other variable names are lower case. `arity` or `n`?

Reply:
Changed to `arity`